### PR TITLE
Fix hanging threads in morsel runtime

### PR DESCRIPTION
**Pull request by pontusmelke**
We were catching an `Exception` instead of a `Throwable` which lead to that for some errors we weren't releasing the latch leading to stuck threads. 

In this PR we also introduce a morsel runtime stress test, add explicit fallaback for _nested-plan expressions_, and enables the morsel runtime in the tck/acceptance suite.

 On top of that we fixed a couple of things in `specs-suite-tools`:
- Use visitor instead of iterator when consuming the results
- Don't generated duplicates when creating blacklists

### CODE REVIEW

File: enterprise/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/MorselRuntimeStressIT.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import org.neo4j.graphdb.Result;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.test.rule.EnterpriseDatabaseRule;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertTrue;
+
+public class MorselRuntimeStressIT
```
Comment (fickludd) [inline enterprise/cypher/cypher/src/test/java/org/neo4j/cypher/internal/javacompat/MorselRuntimeStressIT.java]: Very nice and well needed.

File: enterprise/cypher/morsel-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/vectorized/dispatcher/ParallelDispatcher.scala [hunk 2]
```diff
@@ -91,10 +91,9 @@ class ParallelDispatcher(morselSize: Int, workers: Int, executor: Executor) exte
 
         }
       } catch {
-        case e: Exception =>
+        case e: Throwable =>
```
Comment (fickludd) [inline enterprise/cypher/morsel-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/vectorized/dispatcher/ParallelDispatcher.scala:94]: Just curious.. what was thrown?

**Review by fickludd — APPROVED** [review]
LGTM. Great that we can re-enable regular morsel testing again, even if only in singlethreaded.

File: enterprise/cypher/morsel-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/vectorized/dispatcher/ParallelDispatcher.scala (hunk 2 above)
Comment (pontusmelke) [reply enterprise/cypher/morsel-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/vectorized/dispatcher/ParallelDispatcher.scala:94]: We were using nested pipe expressions from the morsel runtime but nested pipe expressions doesn't have apply defined - it just has `def apply = ???` which will throw an `NotImplementedError` which is an `Error` not an `Exception`

Comment (pontusmelke) [reply enterprise/cypher/morsel-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/vectorized/dispatcher/ParallelDispatcher.scala:94]: and btw it works in parallel mode as well, it is just 3 times slower.

**Pull request merged**